### PR TITLE
Add serialization case option

### PR DESCRIPTION
**Pull request by mpetrunic**

### CODE REVIEW

File: src/backings/structural/container.ts [hunk 1]
```diff
@@ -152,23 +153,29 @@ export class ContainerStructuralHandler<T extends ObjectLike> extends Structural
...
     });
     return value;
   }
-  toJson(value: T): Json {
+  toJson(value: T, options?: IJsonOptions): Json {
     const data = {} as Record<string, Json>;
+    const expectedCase = options ? options.case : null;
     Object.entries(this._type.fields).forEach(([fieldName, fieldType]) => {
-      data[fieldName] = fieldType.toJson(value[fieldName as keyof T]);
+      data[toExpectedCase(fieldName, expectedCase)] = fieldType.toJson(value[fieldName as keyof T]);
```
Comment (wemeetagain) [inline src/backings/structural/container.ts:178]: call to toJson should have options as second arg
we should make sure that all recursively called toJson and fromJson have options passed in (here and in array/vector/list)

**Comment by mpetrunic** [comment]
But we could change some fields that arent declared in ssz?

**Comment by mpetrunic** [comment]
Also this allows further extending with custom primitive serializations etc

File: src/backings/structural/container.ts [hunk 2]
```diff
@@ -152,23 +153,29 @@ export class ContainerStructuralHandler<T extends ObjectLike> extends Structural
...
       throw new Error("Invalid JSON container: expected Object");
     }
     const value = {} as T;
     Object.entries(this._type.fields).forEach(([fieldName, fieldType]) => {
-      if ((data as Record<string, Json>)[fieldName] === undefined) {
+
+      const expectedCase = options ? options.case : null;
+      const expectedFieldName = toExpectedCase(fieldName, expectedCase);
+      if ((data as Record<string, Json>)[expectedFieldName] === undefined) {
         throw new Error(`Invalid JSON container field: expected field ${fieldName} is undefined`);
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply src/backings/structural/container.ts:166]: I think the error should reference the expectedFieldName and not the fieldName?

[the remark this answers is not in the document]
Comment (mpetrunic) [reply src/backings/structural/container.ts:166]: Oh yeah, didn't get your initial comment

**Pull request merged**